app/models/manageiq/providers/ansible_tower/automation_manager/configuration_script_source/api_create.rb

```diff
@@ -0,0 +1,36 @@
+class ManageIQ::Providers::AnsibleTower::AutomationManager::ConfigurationScriptSource
+  module ApiCreate
+    def create_in_provider(manager_id, params)
+      manager = ExtManagementSystem.find(manager_id)
+      project = manager.with_provider_connection do |connection|
+        connection.api.projects.create!(params)
+      end
+
+      # Get the record in our database
+      # TODO: This needs to be targeted refresh so it doesn't take too long
+      EmsRefresh.queue_refresh(manager, nil, true) if !manager.missing_credentials? && manager.authentication_status_ok?
+
+      find_by(:manager_id => manager.id, :manager_ref => project.id)
```

Comment:
Another note: I would not query on ids. You can also do `find_by(:manager => manager, :manager_ref => project.id)` - this lets rails do the mapping to `_id` plus - and this is great for polymorphic stuff - it will fill `manager_type` in case `belongs_to :manager, :polymorphic => true`